homeassistant/components/zwave_js/update.py

```diff
@@ -277,7 +288,14 @@ async def async_added_to_hass(self) -> None:
             )
         )
 
-        self.async_on_remove(async_at_start(self.hass, self._async_update))
+        # Turn state off to start if there is no skipped version
+        if not getattr(self, "_UpdateEntity__skipped_version"):
```

Comment:
Why do we have to worry about details of the base update entity?

Reply:
because before I added this, we were clearing the skipped version when we started up because the state was set to unknown. I am realizing though that there is a better way to handle this

Reply:
this new logic has an issue and now I recall why I went this way - if we restore the latest version and it's higher than the installed version, the user will have the option to install the update. But because we haven't run the update function, if the user clicks install it will fail because we assume that the latest firmware info has been stored but it hasn't in the restoration case. Probably have to implement storing of extra data and retrieve that as part of this process too...

Reply:
OK this is fixed with the latest changes to restoring extra data. Unfortunately, we may have to clear the skipped version for users as part of this transition but I think it's a small price to pay for the end result

Reply:
still needs a test

Reply:
tests added